File: Content/example-code/HelloWorldBasic/csharp/HelloWorldBasic.cs
//version 1.0.2
// %%%% start all
namespace Demos
{
    using System;
    using System.Drawing;
    using Applitools.Selenium;
    using OpenQA.Selenium;
    using OpenQA.Selenium.Chrome;

    public class HelloWorldSimple
    {
        public static void Main(string[] args)
        {     
            var eyes = new Eyes();                                                      // Note 1
            eyes.ApiKey = Environment.GetEnvironmentVariable("MY_APPLITOOLS_API_KEY");  // Note 2
            var innerDriver = new ChromeDriver();                                       // Note 3                                
            var viewportSize = new Size(1024, 768);
            var driver = eyes.Open(innerDriver, 
                    "My Application Name", "My Test Name", viewportSize);               // Note 4
            try
            {
                string website = "https://applitools.com/helloworld";
                driver.Url = website;                                                   // Note 5
                eyes.CheckWindow("initial screen");                                     // Note 6
                Applitools.TestResults result = eyes.Close(false);                      // Note 7
            }
            catch (Exception ex)
            {
            }
            finally
            {
                // If the test was aborted before eyes.Close was called, ends the test as aborted.
                eyes.AbortIfNotClosed();                                                // Note 8
            }
            innerDriver.Quit();                                                         // Note 9
        }
    }
}
// %%%% stop all

File: Content/example-code/HelloWorldBasic/javascript/HelloWorldBasic.js
//version 1.0.2
// %%%% start all
var Eyes = require('eyes.selenium').Eyes;
var Webdriver = require('selenium-webdriver');

function main() {
    var eyes = new Eyes();                                                // Note 1
    var apiKey = process.env.MY_APPLITOOLS_API_KEY;                       
    eyes.setApiKey(apiKey);                                               // Note 2
    var innerDriver = new Webdriver.Builder()                             // Note 3
        .withCapabilities(Webdriver.Capabilities.chrome())
        .build();
    var viewportSize = {width: 1024, height: 768};
    eyes.open(innerDriver,                                                // Note 4  
            'My Application Name', 'My Test Name', viewportSize)
        .then(function (driver){ afterOpen(driver)});
}

function afterOpen(eyes, driver) {
    try {
        var  website = "https://applitools.com/helloworld";
        driver.get(website);                                              // Note 5 
        eyes.checkWindow('Before mouse click');                           // Note 6 
        eyes.close(false)                                                 // Note 7 
        .then(function(testResults) {
            //process results here
        })
    } finally {
      eyes.abortIfNotClosed(); //could add .then if necessary             // Note 8 
    }
    innerDriver.quit(); // Close the browser.                             // Note 9 
}
main();
// %%%% stop all